### PR TITLE
improve: effect select option label naming and grouping

### PR DESCRIPTION
**Pull request by doommm**
Re: #
- https://github.com/cocos/3d-tasks/issues/14323

### Dependent PR

- https://github.com/cocos/engine-extensions/pull/185

### Changelog

* improve effect select option label naming and grouping

image

**After dependent PR and this PR is merged, user should reimport all effects (or delete existing `cretorProject/library` folder) to see the improvement.**

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by arsen2010 — APPROVED** [review]

**Comment by doommm** [comment]
@VisualSJ

File: editor/inspector/assets/material.js [hunk 1]
```diff
@@ -4,6 +4,57 @@ const { materialTechniquePolyfill } = require('../utils/material');
...
 const { join, sep, normalize } = require('path');
 
+/**
+ * 
+ * @param {{name: string; uuid: string}[]} sortedEffects 
+ * @returns html template
+ */
+function renderGroupEffectOptions(sortedEffects) {
+    const groupNames = new Set();
+    const groupNameRE = /^(\w+)\//;
```
Comment (VisualSJ) [inline editor/inspector/assets/material.js]: You can put this outside as a constant

File: editor/inspector/assets/material.js [hunk 2]
```diff
@@ -4,6 +4,57 @@ const { materialTechniquePolyfill } = require('../utils/material');
...
+            if (!groupNames.has(groupName)) {
+                groupNames.add(groupName);
+
+                currGroup = groupName;
+
+                htmlTemplate += `<optgroup label="${groupName}">`;
+            }
+
+            // for option label, remove group name if matched
+            optionLabel = optionLabel.replace(groupNameRE, '');
```
Comment (VisualSJ) [inline editor/inspector/assets/material.js]: Is this path not displayed correctly? What will be displayed if I have files outside the "effect" folder?

Comment (doommm) [reply editor/inspector/assets/material.js]: As required metioned in https://github.com/cocos/3d-tasks/issues/14323

`internal/effects/a` => `a`
`internal/effects/a/b` => `a/b`
`internal/a` => `a`

File: editor/inspector/assets/material.js (hunk 1 above)
Comment (doommm) [reply editor/inspector/assets/material.js]: extract to a const

File: editor/inspector/assets/material.js (hunk 2 above)
Comment (VisualSJ) [reply editor/inspector/assets/material.js]: a !== a?
Isn't that ambiguous?

**Pull request merged**